napari/components/_interaction_box_mouse_bindings.py

```diff
@@ -43,10 +48,151 @@ def __init__(self, viewer, interaction_box_visual):
         self._selected_vertex: int = None
         self._fixed_vertex: int = None
         self._fixed_aspect: float = None
+        self._layer_listening_for_affine = None
         self._viewer = viewer
         self._interaction_box_model = viewer.overlays.interaction_box
         self._interaction_box_visual = interaction_box_visual
+        viewer.layers.events.inserted.connect(self._on_add_layer)
+        viewer.layers.events.removed.connect(self._on_remove_layer)
+        viewer.layers.selection.events.active.connect(self._on_active)
+        viewer.dims.events.order.connect(self._on_dim_change)
+        viewer.dims.events.ndisplay.connect(self._on_ndisplay_change)
+        self._interaction_box_model.events.transform_drag.connect(
+            self._on_transform_change
+        )
         self.initialize_mouse_events(viewer)
+        self.initialize_key_events(viewer)
+
+    def _on_remove_layer(self, event):
+        """Gets called when layer is added and removes event listener"""
+        layer = event.value
+        if hasattr(layer, 'mode'):
+            layer.events.mode.disconnect(self)
+
+    def _on_add_layer(self, event):
+        """Gets called when layer is added and adds event listener to mode change"""
+        layer = event.value
+        if hasattr(layer, 'mode'):
+            layer.events.mode.connect(self._on_mode_change)
+
+    def _on_active(self, event):
+        """Gets called when active layer is changed"""
+        active_layer = event.value
+        if hasattr(active_layer, 'mode') and active_layer.mode == 'transform':
+            self._couple_interaction_box_to_active()
+            self._interaction_box_model.show = True
+            self._layer_affine_event_helper(active_layer)
+        else:
+            self._interaction_box_model.show = False
+
+    def _on_ndisplay_change(self):
+        """Gets called on ndisplay change to disable interaction box in 3D"""
+        if (
+            hasattr(self._viewer.layers.selection.active, 'mode')
+            and self._viewer.layers.selection.active.mode == 'transform'

```

Comment:
Optional: I think this is equivalent
```suggestion
            getattr(self._viewer.layers.selection.active, 'mode', None) == 'transform'
```
and a little easier to read, mostly because of the long reference to `active`.

Alternatively/Additionally you could pull out a part of that named reference at the beginning (e.g. `selection = self._viewer.layers.selection`), if we trust that `selection` will be a reference to some state and not a copy.